Sources/NIOCore/ByteBuffer-views.swift

```diff
@@ -158,6 +158,27 @@ extension ByteBufferView: RangeReplaceableCollection {
         self = ByteBufferView(ByteBuffer())
     }
 
+    /// Reserves enough space in the underlying `ByteBuffer` to store the specified
+    /// number of bytes.
+    ///
+    /// See the documentation for ``ByteBuffer.reserveCapacity(_:)`` for more details.
+    @inlinable
+    public mutating func reserveCapacity(_ minimumCapacity: Int) {
+        self._buffer.reserveCapacity(minimumCapacity)
+    }
+
+    /// Writes a single byte to the underlying `ByteBuffer`.
+    @inlinable
+    public mutating func append(_ byte: UInt8) {
+        self._buffer.writeBytes(CollectionOfOne<UInt8>(byte))
+    }
+
+    /// Writes a sequence of bytes to the underlying `ByteBuffer`.
+    @inlinable
+    public mutating func append<Bytes: Sequence>(contentsOf bytes: Bytes) where Bytes.Element == UInt8 {
+        self._buffer.writeBytes(bytes)
```

Comment:
This does not correctly manipulate `self._range`, and so fails to move `endIndex`. Can you add a test for this function? That test should reveal this issue.

Reply:
done, and done. i’m not familiar with the preconditions of `moveWriterIndex(to:)`, so i hope i am using it correctly…